doc/source/dev/roadmap-detailed.rst

```diff
@@ -133,15 +151,13 @@ This module is in good shape.
 
 integrate
 `````````
-Needed for ODE solvers:
-
-- Documentation is pretty bad, needs fixing
-- A new ODE solver interface  (``solve_ivp``) was added in SciPy 1.0.0.
-  In the future we can consider (soft-)deprecating the older API.
-
-The numerical integration functions are in good shape.  Support for integrating
-complex-valued functions and integrating multiple intervals (see `gh-3325
-<https://github.com/scipy/scipy/issues/3325>`__) could be added.
+- Complete the feature set of the new ``cubature`` function, and add an interface
+  tailored to one-dimensional integrals.
+- Migrate functions for generating quadrature rule points and weights from `special`,
+  improve their reliability, and add support for other important rules.
+- Complete the feature set of ``solve_ivp``, including all functionality of the
```

Comment:
A couple of possible improvements:
- The methods we have are quite outdated eg. `rk45` should be replaced with `tsit5` see https://docs.sciml.ai/DiffEqDocs/stable/solvers/ode_solve/#Translations-from-MATLAB/Python/R
- Investigate fast/low precision exponentiation for the adaptive step size calculation 
- Investigate changing the step size control from an integral controller to a pi controller, see https://docs.sciml.ai/DiffEqDocs/stable/extras/timestepping/
- Accept low level callables
See https://www.stochasticlifestyle.com/comparison-differential-equation-solver-suites-matlab-r-julia-python-c-fortran/ for more areas of improvement

Reply:
It would also be great if SciPy had a dae solver

Reply:
I wrote this part mostly from the perspective of letting go of the old interface. Personally, I'd be very interested in working on Array API compatible ODE/DAE solvers, but again I think there needs to be a study of what needs are not met by the rest of the ecosystem.

Reply:
discussion on DAE in https://github.com/scipy/scipy/pull/13068

Reply:
> Looks like you have done some research - does SciPy need to step up in this department, or are other Python packages active and better places for the best new methods?
I try to keep an eye on what's happening in the Julia world, they have brought some really innovation to methods that haven't seen any innovation in a long time. 

It depends how flexible your requirements are:
- https://github.com/SciML/diffeqpy wraps differentialequations.jl and is great but requires you to have Julia install
- https://github.com/patrick-kidger/diffrax looks good although I have only looked at a surface level, only works for jax